pkg/api/types.go

```diff
@@ -296,6 +296,7 @@ type HostedMasterProfile struct {
 	// Not used during PUT, returned as part of GETFQDN
 	FQDN      string `json:"fqdn,omitempty"`
 	DNSPrefix string `json:"dnsPrefix"`
+	Subnet    string `json:"subnet"`
```

Comment:
IMO a comment is needed here that this is not really the subnet of the master/control plane but actually used during template generation to assign the subnet to agents.